### PR TITLE
Add Alluxio FUSE SDK tarball generation PR

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?
PR 5 of Alluxio FUSE SDK https://github.com/Alluxio/alluxio/pull/16284
This PR contains the logic to create a Alluxio Fuse SDK tarball

### Why are the changes needed?

To be able to create the separate Alluxio FUSE SDK tarball

### Does this PR introduce any user facing changes?

NA

### CODE REVIEW

**Review by beinan — APPROVED** [review]
lgtm

File: dev/scripts/src/alluxio.org/build-distribution/cmd/common.go
Comment (Xenorith) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/common.go:82]: how did you determine to separate these jars from the original list? seems like some but not all ufs jars are moved

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go
Comment (Xenorith) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go:325]: i wonder how much of the tarball generation codepath is share with fuse vs not. is it worth combining into one codepath...?

**Review by Xenorith — COMMENTED** [review]
would like to take a closer look at this PR before it is merged

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go
Comment (LuQQiu) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go:325]: Sort of hard to decide whether to share or not, share code is non-trivial and non-shared code is also non-trivial

File: dev/scripts/src/alluxio.org/build-distribution/cmd/common.go
Comment (LuQQiu) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/common.go:82]: This is from the initial discussion with @beinan and @apc999 about what UFS to include first and give those UFSes for users to try out and for benchmarking. This PR is merged to fuse-sdk branch first and the UFSes are not the final decision.  For final product delivery, still need further discussion of what UFSes to include and possibly minimize to S3/Azure/GCS only.

**Comment by LuQQiu** [comment]
@Xenorith any concern?

File: dev/scripts/src/alluxio.org/build-distribution/cmd/common.go
Comment (Xenorith) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/common.go]: avoid copying the module part by keeping a variable listing the keys of the map that you want as fuse ufs modules

File: dev/scripts/src/alluxio.org/build-distribution/cmd/common.go
Comment (Xenorith) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/common.go:94]: libJars -> coreLibJars + additionalLibJars

File: dev/scripts/src/alluxio.org/build-distribution/cmd/flags.go
Comment (Xenorith) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/flags.go]: make the initialization of the opts struct explicit declare what these values are instead of hiding them within the logic

File: dev/scripts/src/alluxio.org/build-distribution/cmd/flags.go
Comment (Xenorith) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/flags.go:68]: this was a bug since it was only iterating through the additional jars, missing the core ones

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-fuse-tarball.go
Comment (Xenorith) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/generate-fuse-tarball.go]: wrapping the `defaultModules()` means that you would not be including the hadoop-3.3 module since it's default flag is false. i assume that is the behavior you want

**Comment by Xenorith** [comment]
there are more refactors i'd like to do but they are not relevant to the fuse additions. so many places we update global variables so it is a mess to keep track of what has happened

**Comment by LuQQiu** [comment]
> there are more refactors i'd like to do but they are not relevant to the fuse additions. so many places we update global variables so it is a mess to keep track of what has happened

When i modify the script... i think the same as well... so many places want to refactor but worry about breaking 2.9 lol

File: dev/scripts/src/alluxio.org/build-distribution/cmd/common.go
Comment (LuQQiu) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/common.go]: Yeah, copy all the map fields do not make sense

File: dev/scripts/src/alluxio.org/build-distribution/cmd/flags.go
Comment (LuQQiu) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/flags.go:68]: Good catch!

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-fuse-tarball.go
Comment (LuQQiu) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/generate-fuse-tarball.go]: change to use hadoop-3.2 instead of 3.3 since it looks like common (it's marked as default)

File: dev/scripts/src/alluxio.org/build-distribution/cmd/common.go
Comment (LuQQiu) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/common.go]: another way is to add a field to the module
all ufs modules - default ufs modules - core ufs modules
so that there is hierachy, but will discuss to see whether we actually need hadoop jars

**Review by Xenorith — APPROVED** [review]

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**
